### ISSUE TITLE
development.readium.divshot.io 'packed epub' not working?

### ISSUE THREAD

**Issue by davedmiller** [readium/readium-js-viewer#381]
I was just looking at the example web site development.readium.divshot.io and when I click on the 'packed epub' example, nothing seems to happen.  Readium comes up, but all I get is a spinner.  I am running on Windows 8.1 using the Chrome Browser Version 44.0.2403.130 m.

Is this correct behavior?

**Comment by rkwright** [comment]
@davedmiller Yes, this should work, with some caveats.  This is a bug, AFAIK.

Closing as a duplicate of #382

**Comment by danielweck** [comment]
This is a known issue with DivShot.io HTTP headers, see https://github.com/divshot/superstatic/issues/169

Note that this works fine with Surge.sh http://readium.surge.sh/?epub=http://development.readium.divshot.io/epub_content/epubReadingSystem.epub
(however, this cloud hosting service does not support HTTP CORS headers in the way DivShot does, so there are pros and cons with each platform)

**Comment by danielweck** [comment]
UPDATE: I removed the part about Surge.sh, as the zipped EPUB isn't hosted there anymore, so this can't be used to demonstrate successful load. However, this example works: http://development.readium.divshot.io/?epub=http://www.daisy.org/projects/tobi/readium/epub_content/internal_link.epub
(as you can see, the zipped EPUB loads fine, because the daisy.org server provides adequate headers)

**Comment by danielweck** [comment]
@rkwright as this is a DivShot.io bug, it would make sense to close this issue. However, on several occasions I mentioned the idea of using a cloud / hosting service for the Readium app itself, possibly alongside a separate host for the EPUB content (which can be quite large because of audio/video assets, EPUB-testsuite, etc.). Thus why I implemented HTTP CORS support for DivShot.io, daisy.org (my own corporate server), and looked at Surge.sh as a possible alternative / complement (there are plenty of other choices though...some with adequate "freemium" pricing models).

The bottom line is: for as long as the DivShot.io HTTP header is missing, Readium cloud reader will fail to load zipped EPUBs hosted at DivShot. The app will load EPUBs fine if they are hosted on properly-configured servers though. For example: http://development.readium.divshot.io/?epub=http://www.daisy.org/projects/tobi/readium/epub_content/internal_link.epub

**Comment by danielweck** [comment]
Fixed here: https://github.com/readium/readium-js/commit/6ec3b39ba464712c1b600d6245ef9b74b8ddf1e2

This fallback solution uses the new Blob fetch method introduced here:
https://github.com/readium/readium-js/pull/115

Test:
http://development.readium.divshot.io/?epub=epub_content%2Finternal_link.epub

HTTP CORS test:
http://readium.surge.sh/?epub=http://development.readium.divshot.io/epub_content/internal_link.epub

**Comment by danielweck** [comment]
Ah, what works on my development environment doesn't on DivShot.io...so I am not closing this issue just yet.

**Comment by danielweck** [comment]
Ah, now works!

Same-origin test:
http://development.readium.divshot.io/?epub=epub_content%2Finternal_link.epub

HTTP CORS tests:
http://readium.surge.sh/?epub=http://development.readium.divshot.io/epub_content/internal_link.epub

http://www.daisy.org/projects/tobi/readium/?epub=http://development.readium.divshot.io/epub_content/internal_link.epub

**Comment by danielweck** [comment]
Document which surveys a bunch of existing HTTP servers / hosting services (includes CORS review):
https://docs.google.com/document/d/1dFDrmStg-bRv4n1wFxnDeTnMuaFRlmxk3jw4gEwfDDo
